### PR TITLE
SOV-2129: Pull Zero fee rates directly from contracts

### PR DESCRIPTION
**Pull request by pietro-maximoff**
https://sovryn.atlassian.net/browse/SOV-2129

### CODE REVIEW

**Review by creed-victor — CHANGES_REQUESTED** [review]
I think, instead of having a hook retrieving fees you should move the call to the util function like it was used previously (ethers.getFees()), ex:
`getZeroFess(): { borrowingRate: Decimal; slippage: Decimal }`

Value also could be cached.
With your approach we will need to always call a hook and then pass values as an arguments/props, which is not really necessary as these values are almost static

File: apps/frontend/src/app/5_pages/ZeroPage/utils/trove-manager.ts [hunk 1]
```diff
@@ -52,16 +89,15 @@ export const adjustTrove = async (
 
   const { ethers } = await getZeroProvider();
 
-  const [trove, fees] = await Promise.all([
+  const [trove] = await Promise.all([
     ethers.getTrove(address),
     borrowZUSD && ethers.getFees(),
   ]);
```
Comment (creed-victor) [inline apps/frontend/src/app/5_pages/ZeroPage/utils/trove-manager.ts]:
```suggestion
  const [trove, fees] = await Promise.all([
    ethers.getTrove(address),
    borrowZUSD && getLiquityBaseParams(),
  ]);
```

File: apps/frontend/src/app/5_pages/ZeroPage/utils/trove-manager.ts [hunk 2]
```diff
@@ -52,16 +89,15 @@ export const adjustTrove = async (
...
 
-  const borrowingRate = fees?.borrowingRate();
-  const finalTrove = trove.adjust(normalized, borrowingRate);
+  const { minBorrowingFeeRate, maxBorrowingFeeRate } =
+    await getLiquityBaseParams();
```
Comment (creed-victor) [inline apps/frontend/src/app/5_pages/ZeroPage/utils/trove-manager.ts]:
```suggestion
```

File: apps/frontend/src/app/5_pages/ZeroPage/utils/trove-manager.ts [hunk 3]
```diff
@@ -52,16 +89,15 @@ export const adjustTrove = async (
...
 
-  const maxBorrowingRate =
-    borrowingRate?.add(defaultBorrowingRateSlippageTolerance) ?? Decimal.ZERO;
+  const finalTrove = trove.adjust(normalized, minBorrowingFeeRate);
```
Comment (creed-victor) [inline apps/frontend/src/app/5_pages/ZeroPage/utils/trove-manager.ts]:
```suggestion
  const finalTrove = trove.adjust(normalized, fees.minBorrowingFeeRate);
```

File: apps/frontend/src/app/5_pages/ZeroPage/utils/trove-manager.ts [hunk 4]
```diff
@@ -71,7 +107,7 @@ export const adjustTrove = async (
     value: value.hex,
     fn: token === SupportedTokens.dllr ? 'adjustNueTrove' : 'adjustTrove',
     args: [
-      maxBorrowingRate.hex,
+      maxBorrowingFeeRate,
```
Comment (creed-victor) [inline apps/frontend/src/app/5_pages/ZeroPage/utils/trove-manager.ts]:
```suggestion
      fees.maxBorrowingFeeRate,
```

File: apps/frontend/src/app/5_pages/ZeroPage/hooks/useLiquityBaseParams.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        const [getMinBorrowingFee, getMaxBorrowingFee] = await Promise.all([
+          contract.BORROWING_FEE_FLOOR(),
+          contract.MAX_BORROWING_FEE(),
+        ]);
+        const minBorrowingFeeRate = Decimal.fromBigNumberString(
+          String(getMinBorrowingFee),
+        );
+        const maxBorrowingFeeRate = Decimal.fromBigNumberString(
+          String(getMaxBorrowingFee),
+        );
```
Comment (creed-victor) [inline apps/frontend/src/app/5_pages/ZeroPage/hooks/useLiquityBaseParams.ts]:
```suggestion
        const { minBorrowingFeeRate, maxBorrowingFeeRate } = getLiquityBaseParams();
```

File: apps/frontend/src/app/5_pages/ZeroPage/ZeroPage.tsx [hunk 6]
```diff
@@ -128,10 +130,13 @@ const ZeroPage: FC<ZeroPageProps> = ({ deferred: [price, fees] }) => {
...
-    () => decimalic(fees?.borrowingRate()?.toString()),
-    [fees],
-  );
+  useEffect(() => {
+    const fetchBorrowingRate = async () => {
+      const params = await getLiquityBaseParams();
+      setBorrowingRate(decimalic(params.minBorrowingFeeRate.toString()));
+    };
+    fetchBorrowingRate();
+  }, []);
```
Comment (creed-victor) [inline apps/frontend/src/app/5_pages/ZeroPage/ZeroPage.tsx]:
```suggestion
const { minBorrowingFeeRate } = useLiquityBaseParams();
const borrowingRate = useMemo(() => decimalic(minBorrowingFeeRate), [minBorrowingFeeRate]);
```

File: apps/frontend/src/app/5_pages/ZeroPage/ZeroPage.tsx [hunk 7]
```diff
@@ -58,6 +59,7 @@ const ZeroPage: FC<ZeroPageProps> = ({ deferred: [price, fees] }) => {
   const [trove, setTrove] = useState<UserTrove>();
   const [collateralSurplusBalance, setCollateralSurplusBalance] =
     useState<Decimal>();
+  const [borrowingRate, setBorrowingRate] = useState(Decimal.ZERO);
```
Comment (creed-victor) [inline apps/frontend/src/app/5_pages/ZeroPage/ZeroPage.tsx]:
```suggestion
```

**Review by creed-victor — CHANGES_REQUESTED** [review]

File: apps/frontend/src/app/5_pages/ZeroPage/utils/trove-manager.ts [hunk 8]
```diff
@@ -37,7 +73,7 @@ export const openTrove = async (
   return {
     value: value.hex,
     fn: token === SupportedTokens.dllr ? 'openNueTrove' : 'openTrove',
-    args: [maxBorrowingRate.hex, borrowZUSD.hex, ...hints],
+    args: [fees?.maxBorrowingFeeRate, borrowZUSD.hex, ...hints],
```
Comment (creed-victor) [inline apps/frontend/src/app/5_pages/ZeroPage/utils/trove-manager.ts]:
```suggestion
    args: [fees?.maxBorrowingFeeRate.toHexString(), borrowZUSD.hex, ...hints],
```
I think it's decimal now, so make sure you are passing hex to contract

File: apps/frontend/src/app/5_pages/ZeroPage/utils/trove-manager.ts [hunk 9]
```diff
@@ -71,7 +103,7 @@ export const adjustTrove = async (
     value: value.hex,
     fn: token === SupportedTokens.dllr ? 'adjustNueTrove' : 'adjustTrove',
     args: [
-      maxBorrowingRate.hex,
+      fees?.maxBorrowingFeeRate,
```
Comment (creed-victor) [inline apps/frontend/src/app/5_pages/ZeroPage/utils/trove-manager.ts]:
```suggestion
      fees?.maxBorrowingFeeRate.toHexString(),
```

**Review by creed-victor — CHANGES_REQUESTED** [review]

**Review by creed-victor — APPROVED** [review]

**Review by tiltom — APPROVED** [review]

**Pull request merged**